src/main/java/org/myrobotlab/service/config/ServiceConfig.java

```diff
@@ -0,0 +1,18 @@
+package org.myrobotlab.service.config;
+
+public class ServiceConfig {
+  
+  public String name;
+  public String type;   
+  public Config config;
+  
+  public ServiceConfig() {    
+  }
+  
+  public ServiceConfig(String name, String type, Config config) {
+    this.name = name;
+    this.type = type;
+    this.config = config;
```

Comment:
remove "config" as a member here.. subclasses should implement service specific configs.